Add porting guide and other docs (#3)
nrfconnect/nrf70-bm: nrf70_bm_lib/docs/source/nrf70_bm_porting_guide.rst

.. _nrf70_bm_porting_guide:

> nRF70 BM Porting Guide

Reviewer: ```suggestion
nRF70 Series BM Driver Porting Guide
```

======================

> The nRF70 BM library is designed to be portable across different platforms.

Reviewer: Same comment as above. I suggest we use the term Library for the RTOS-agnostic part and Driver for the whole solution.

This guide provides an overview of the nRF70 BM library and the steps to port the library to a new platform.

nRF70 driver threading model
****************************

The nRF70 driver uses a simple threading model to interact with the nRF70.

* *Interrupt context*: For handling interrupts from the nRF70. Interrupt context is used to schedule tasklets to handle the nRF70.
  A `host IRQ` GPIO pin raises an interrupt to the host MCU when the nRF70 has an event to report.
  The interrupt handler reads the event from the nRF70 and schedules a tasklet to handle the event.
* *Tasklet context*: Tasklets do the actual work of interacting with the nRF70, reding and writing commands and events.

>   RX tasklet: The RX tasklet reads the event data from the nRF70 and hands over to the FMAC callbacks.

Author: No, there isn't, ISR context is only for events (RX and TX DONE both).

* *Caller context*: The caller context is the context in which the FMAC APIs are called.
  The caller context is responsible for calling the FMAC APIs.

Zephyr shim reference
**********************

The `nrf70_zephyr_shim` directory contains a reference implementation for Zephyr.
This reference implementation can be used as a guide to port the library to other platforms.

> This reference implementation has below directory structure:

Reviewer: ```suggestion
This reference implementation is structured in multiple source directories as follows :
```

.. code-block:: none

>     nrf70_zephyr_shim/     # Zephyr shim for nRF70, reference for customer boards

Reviewer: ```suggestion
    nrf70_zephyr_shim/     # Zephyr shim for nRF70, reference for third-party host platforms
```

      include/             # Include directory
      source/              # Source directory
        bus/               # Bus interface
        os/                # OS shim
        platform/          # Platform specific files
      CMakeLists.txt       # CMake build file```

The key components of the Zephyr shim reference implementation are:

* *os*: Contains the OS shim implementation for Zephyr.
* *bus*: Contains the bus interface implementation for SPI.
* *platform*: Contains the platform specific files, and has an RPU (Radio Processing Unit) abstraction layer that interacts with the nRF70

>   either through QSPI or SPI. The platform also uses Zephyr GPIO APIs to manage GPIO pins of nRF70.

Reviewer: should we keep QSPI in the guide?

Author: I left it here just in case a platform has support for QSPI even with EK.

Zephyr functionality used in the reference implementation:
**********************************************************

> Below are the Zephyr functionalities used in the reference implementation, most of these functionalities are standard OS primitives and can be easily ported to other platforms.

Reviewer: ```suggestion
The reference implementation of the BM Driver for the Zephyr RTOS uses build-time tools as well as standard OS primitives; these elements require porting effort when integrating the driver to a third-party (non-Zephyr) platform.  
```

Reviewer: Let's split the list below in 3 groups
- OS primitives
   - Heap
   - Sleep, Delay
   - Workqueue
   - Semaphores
   - Lists 
- driver model
   - SPI 
   - GPIO
- build-time tools
  - DTS
